Detectors/TRD/calibration/macros/plotVdriftExB.C

```diff
@@ -0,0 +1,133 @@
+
+// Download the values and populate the map.
+void ccdbDownload(unsigned int runNumber, std::string ccdb, timePoint timeOffset)
+{
+  auto& ccdbMgr = o2::ccdb::BasicCCDBManager::instance();
+  ccdbMgr.setURL("http://alice-ccdb.cern.ch/");
+  auto runDuration = ccdbMgr.getRunDuration(runNumber);
+  std::map<std::string, std::string> md;
+  md["runNumber"] = std::to_string(runNumber);
+  const auto* grp = ccdbMgr.getSpecific<o2::parameters::GRPECSObject>("GLO/Config/GRPECS", (runDuration.first + runDuration.second) / 2, md);
+  grp->print();
+  const auto startTime = grp->getTimeStart();
+  const auto endTime = grp->getTimeEnd();
+  ccdbMgr.setURL(ccdb);
+
+  for (timePoint time = startTime; time < endTime; time += timeOffset) {
+    ccdbMgr.setTimestamp(time);
+    std::cout << "Getting vDriftExB for " << time << std::endl;
+    auto calVdriftExB =
+      ccdbMgr.get<o2::trd::CalVdriftExB>("TRD/Calib/CalVdriftExB");
+    for (int iDet = 0; iDet < 540; ++iDet) {
+      vmap[iDet].push_back(std::make_tuple(
+        calVdriftExB->getVdrift(iDet), calVdriftExB->getExB(iDet), static_cast<int>(time - startTime)));
+    }
+  }
+}
+
+// Find all good chambers
+// 'Good' is a chamber when it does not contain the defautl values.
+void find_good()
+{
+  for (int iDet = 0; iDet < 540; ++iDet) {
+    good[iDet] = true;
+    if (vmap[iDet].size() == 0) { // No data for detector
+      good[iDet] = false;
+      continue;
+    }
+    for (auto& e : vmap[iDet]) {
+      if (std::get<0>(e) == 1.0 && std::get<1>(e) == 0.0) {
+        good[iDet] = false;
+      }
+    }
+  }
+}
+
+// Print out all 'good' chambers.
+void print_good()
+{
+  printf("Good Chamber(s):\n");
+  for (int iDet = 0; iDet < 540; ++iDet) {
+    if (!good[iDet])
+      continue;
+
+    printf("%i;", iDet);
+  }
+  printf("\n");
+}
+
+std::unique_ptr<TMultiGraph> draw(int i)
+{
+  auto mg = std::make_unique<TMultiGraph>();
+  auto g_vDrift = new TGraph();
+  g_vDrift->SetLineColor(kBlue);
+  auto g_ExB = new TGraph();
+  g_ExB->SetLineColor(kRed);
+  for (auto& e : vmap[i]) {
+    g_vDrift->AddPoint(std::get<2>(e), std::get<0>(e));
+    g_ExB->AddPoint(std::get<2>(e), std::get<1>(e));
+  }
+
+  g_vDrift->Fit("pol0", "q");
+  g_ExB->Fit("pol0", "q");
+  mg->Add(g_vDrift);
+  mg->Add(g_ExB);
+  mg->SetName(Form("vDrift_ExV_%d", i));
+  mg->SetTitle(Form("VDrift (=blue) and ExB (=red) - Chamber %d (%s)", i, (good[i]) ? "GOOD" : "BAD"));
+  mg->GetXaxis()->SetTitle("time since RunStart (ms)");
+
+  return std::move(mg);
+}
+
+// Plot the calibration values for a particular duration.
+// The times must be given in unix epoch ms format.
+// The default Run is 523677 for the test ccdb.
+// The offset for the next calibration is 15 Minutes.
+void plotVdriftExB(unsigned int runNumber = 523677, std::string ccdb = "http://ccdb-test.cern.ch:8080", timePoint timeOffset = 899991)
```

Comment:
Why do you not take 15 minutes (900000) for the `timeOffset`? Also something like `queryInterval` might be more intuitive, but this is up to you

Reply:
I do not recall the reason for this exactly but I think in the ccdb the offsets are like this.
But changing this is works just as well.
Renaming is probably better.
_There are only two hard things in Computer Science: cache invalidation, naming things and off by one errors._